catalog/resource_grants.go

```diff
@@ -316,6 +328,9 @@ func ResourceGrants() common.Resource {
 			}
 			var grants PermissionsList
 			common.DiffToStructPointer(d, s, &grants)
+			if firstPassOfForcedNew(d) {
```

Comment:
Testing for this is covered in integration test below. 

We need to add support for testing CustomizeDiff before we can unit test the 2phases flow of forced new as part of unit test.